packages/issuer-token-endpoint/lib/IssuerTokenEndpoint.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  AccessTokenResponse,
+  Alg,
+  CNonceState,
+  CredentialOfferState,
+  EXPIRED_PRE_AUTHORIZED_CODE,
+  getNumberOrUndefined,
+  GrantType,
+  INVALID_PRE_AUTHORIZED_CODE,
+  IStateManager,
+  Jwt,
+  JWT_SIGNER_CALLBACK_REQUIRED_ERROR,
+  JWTSignerCallback,
+  NONCE_STATE_MANAGER_REQUIRED_ERROR,
+  PIN_NOT_MATCH_ERROR,
+  PIN_NOT_MATCHING_ERROR,
+  PIN_VALIDATION_ERROR,
+  PRE_AUTH_CODE_LITERAL,
+  PRE_AUTHORIZED_CODE_REQUIRED_ERROR,
+  STATE_MANAGER_REQUIRED_ERROR,
+  TokenErrorResponse,
+  Typ,
+  UNSUPPORTED_GRANT_TYPE_ERROR,
+  USER_PIN_NOT_REQUIRED_ERROR,
+  USER_PIN_REQUIRED_ERROR,
+} from '@sphereon/openid4vci-common'
+import express, { NextFunction, Request, Response, Router } from 'express'
+import { v4 } from 'uuid'
+
+const router = express.Router()
+
+interface ITokenEndpointOpts {
+  tokenPath?: string
+  interval?: number
+  cNonceExpiresIn?: number
+  tokenExpiresIn?: number
+  stateManager?: IStateManager<CredentialOfferState>
+  nonceStateManager?: IStateManager<CNonceState>
+  accessTokenSignerCallback?: JWTSignerCallback
+}
+
+export const tokenRequestEndpoint = (opts?: ITokenEndpointOpts): Router => {
+  const tokenPath = opts?.tokenPath ?? process.env.TOKEN_PATH ?? '/token'
+  const interval = opts?.interval ?? getNumberOrUndefined(process.env.INTERVAL) ?? 300000
+  const cNonceExpiresIn = opts?.cNonceExpiresIn ?? getNumberOrUndefined(process.env.C_NONCE_EXPIRES_IN) ?? 300000
+  const tokenExpiresIn = opts?.tokenExpiresIn ?? getNumberOrUndefined(process.env.TOKEN_EXPIRES_IN) ?? 300000
+  if (!opts?.accessTokenSignerCallback) {
+    throw new Error(JWT_SIGNER_CALLBACK_REQUIRED_ERROR)
+  }
+  if (!opts?.stateManager) {
+    throw new Error(STATE_MANAGER_REQUIRED_ERROR)
+  }
+  if (!opts?.nonceStateManager) {
+    throw new Error(NONCE_STATE_MANAGER_REQUIRED_ERROR)
+  }
+  router.post(
+    tokenPath,
+    handleHTTPStatus400({ stateManager: opts.stateManager }),
+    handleTokenRequest({
+      accessTokenSignerCallback: opts.accessTokenSignerCallback,
+      nonceStateManager: opts.nonceStateManager,
+      cNonceExpiresIn,
+      interval,
+      tokenExpiresIn,
+    })
+  )
+  return router
+}
+
+const generateAccessToken = async (
+  opts: Required<Pick<ITokenEndpointOpts, 'accessTokenSignerCallback' | 'tokenExpiresIn'> & { state: string }>
+): Promise<string> => {
+  const issuanceTime = new Date()
+  const jwt: Jwt = {
+    header: { typ: Typ.JWT, alg: Alg.ES256 },
+    payload: { iat: issuanceTime.getTime(), exp: opts.tokenExpiresIn, iss: opts.state },
```

Comment:
What is the state doing in the iss field (that is the field for an identity of the issuer?

Reply:
Fixed!